pages/api/messages.js

```diff
@@ -9,45 +22,102 @@ const allowedFields = [
 ];
 
 export default function handler(req, res) {
-  const { fields, fromUserId } = req.query;
-
-  // <--- this part filters by field ---->
-  let filteredThreads = messagesData.map((thread) => {
-    // add last message to response, to make accessing the datetime of the last message and summary easier
-    thread.last_message = thread.messages[thread.messages.length - 1];
-    return thread;
-  });
-
-  if (fields) {
-    const filteredFields = fields
-      .split(',')
-      .filter((field) => allowedFields.includes(field));
-
-    if (filteredFields.length > 0) {
-      filteredThreads = messagesData.map((thread) => {
-        const filteredThread = {};
-
-        filteredFields.forEach((key) => {
-          filteredThread[key] = thread[key];
-        });
+  if (req.method === 'POST') {
+    const { action } = req.query;
+    const { index, fromUserId, like, text } = req.body;
 
-        return filteredThread;
-      });
+    if (!fromUserId) {
+      return res
+        .status(400)
+        .send('Bad request. fromUserId needs to be provided');
+    }
+    if (typeof index !== 'number') {
+      return res.status(400).send('Bad request. index needs to be a number');
     }
-  }
-  // <!--- done filtering by field ---->
 
-  // <--- this part filters by fromUserId ---->
-  if (!fromUserId) {
-    return res.status(200).json(filteredThreads);
+    switch (action) {
+      case 'messageLike': {
+        const thread = messagesData.find(
+          (thread) => thread.from_user_id === fromUserId
+        );
+        const message = thread.messages[index];
+        message.is_liked_by_user = like;
+        syncMessageFile();
+        return res.status(200).json({
+          ok: true,
+          message: `Message is now ${like ? 'liked' : 'unliked'}`,
+        });
+      }
+      case 'deleteMessage': {
+        const thread = messagesData.find(
+          (thread) => thread.from_user_id === fromUserId
+        );
+        thread.messages.splice(index, 1);
+        syncMessageFile();
+        return res
+          .status(200)
+          .json({ ok: true, message: 'Message is now deleted' });
+      }
+      case 'submitMessage': {
+        const thread = messagesData.find(
+          (thread) => thread.from_user_id === fromUserId
+        );
+        thread.messages.push({
+          message_body: text,
+          sent_on: new Date().toISOString(),
+          is_liked_by_user: false,
+          direction: 'sent',
+        });
+        syncMessageFile();
+        return res
+          .status(200)
+          .json({ ok: true, message: 'Message is now Added' });
+      }
+      default: {
+        return res.status(400).send(`Bad request. Bad action type: ${action}`);
```

Comment:
I just realized that sending the wrong action back to the user is a security threat. Let's remove this.
```suggestion
        return res.status(400).send(`Bad request. Bad action type.`);
```